### PR TITLE
feat: db migrations

### PR DESCRIPTION
**Pull request by nduchak**
add db-migration command to cli

### CODE REVIEW

**Comment by nduchak** [comment]
Closes #1

File: src/cli/db-migration.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
+import fs from 'fs'
```
Comment (AuHau) [inline src/cli/db-migration.ts:1]: This command should replace the `db-sync` command. Not sure if this command should be renamed to `db-sync` for backward compatibility or just remove the `db-sync` and keep  `db-migrations` 🤔 I will leave that to your decision.

File: src/cli/db-migration.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      exclusive: ['down', 'generate']
+    }),
+    down: flags.boolean({
+      char: 'd',
+      description: 'Undo db migrations',
+      exclusive: ['up', 'generate']
+    }),
+    generate: flags.string({
+      char: 'd',
+      description: 'Generate migrations',
```
Comment (AuHau) [inline src/cli/db-migration.ts:43]: This description is not helpful at all. It requires some string, but I don't know what I should put there? I thought that it is just a boolean flag originally.

Comment (AuHau) [reply src/cli/db-migration.ts:43]: I also thought that this generate actual migration and not just a template. Maybe this info should be given as well.

File: migrations/index.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    return this.umzugIns.on
+  }
+
+  // eslint-disable-next-line require-await
+  async pending (): Promise<Umzug.Migration[]> {
+    return this.umzugIns.pending().catch(e => {
+      if (e.code === 'ENOENT') {
+        return []
+      }
+      throw e
```
Comment (AuHau) [inline migrations/index.ts:48]: This is not in `async` function, so you should not use `throw` here.

File: package.json [hunk 4]
```diff
@@ -64,6 +64,7 @@
     "@rsksmart/rns-auction-registrar": "1.0.2",
     "@rsksmart/rns-reverse": "^1.0.2",
     "@rsksmart/rns-rskregistrar": "^1.2.1",
+    "@types/umzug": "^2.2.3",
```
Comment (AuHau) [inline package.json:67]: This should be in `devDependencies`

File: src/sequelize.ts [hunk 5]
```diff
@@ -51,12 +52,16 @@ export function BigNumberStringType (propName: string): Partial<ModelAttributeCo
...
-export default function (app: Application): void {
+export default async function (app: Application): Promise<void> {
   const sequelize = sequelizeFactory()
   const oldSetup = app.setup
 
   app.set('sequelize', sequelize)
 
+  // Run migration
+  logger.info('Run DB Migrations')
+  await DbMigration.getInstance(sequelize).up()
```
Comment (AuHau) [inline src/sequelize.ts:63]: You should not run migrations automatically! 
There should be a check here if there are no pending migrations and if so then throw an error, exit and let the user handle this situation with the `db-migration` command!

**Review by AuHau — CHANGES_REQUESTED** [review]
Hmm there are some changes needed here that I missed in the previous PR in Pinner. We should backport those changes to Pinner as well!

Also, I believe that this PR should contain the first migration that will bootstrap the DB into the current state!

File: src/cli/db-migration.ts (hunk 1 above)
Comment (nduchak) [reply src/cli/db-migration.ts:1]: Ok. Probably I remove db-sync, and leave migration command as it is

File: src/cli/db-migration.ts (hunk 2 above)
Comment (nduchak) [reply src/cli/db-migration.ts]: Done

File: migrations/index.ts (hunk 3 above)
Comment (nduchak) [reply migrations/index.ts]: Done

File: package.json (hunk 4 above)
Comment (nduchak) [reply package.json]: Done

File: src/sequelize.ts [hunk 6]
```diff
@@ -51,12 +52,16 @@ export function BigNumberStringType (propName: string): Partial<ModelAttributeCo
   }
 }
 
-export default function (app: Application): void {
+export default async function (app: Application): Promise<void> {
```
Comment (AuHau) [inline src/sequelize.ts:55]: Interesting that linter doesn't yell about not using `await` 🤔

Comment (nduchak) [reply src/sequelize.ts:55]: Had `await` here: https://github.com/rsksmart/rif-marketplace-cache/pull/269/files/a34fa72a7815c31bd81e6a900e46e2f4dcf090f6#diff-f39990cee8f2282dd02df78076ac6a13R61

Comment (AuHau) [reply src/sequelize.ts:55]: Oh missed that one 😅

**Review by AuHau — APPROVED** [review]
LGTM! But maybe I would wait with merge until the `release/mainnet` is done and then regenerated the initial migration and start versioning DB after that 🤔

**Review by AuHau — APPROVED** [review]
LGTM!

**Pull request merged**